.github/workflows/build-and-deploy-staging.yaml

```
@@ -0,0 +1,79 @@
+    runs-on: ubuntu-latest
+    steps:       
+      - name: Create kube config
+        run: |
+          mkdir -p $HOME/.kube/
+          mkdir -p ~/.aws
+          aws configure set aws_access_key_id ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws configure set aws_secret_access_key ${{secrets.AWS_SECRET_ACCESS_KEY }}
+          aws configure set default.region ${{ secrets.AWS_DEFAULT_REGION }}
+          aws eks update-kubeconfig --region us-east-1 --name gateway-go
+      - name: Checkout infra tools repo
+        uses: actions/checkout@v3
```

actionlint (1.7.7)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)